### PR TITLE
Fix infinite scroll on p2

### PR DESCRIPTION
**Pull request by scruffian**
#9120 introduced a bug into infinite scroll which meant that the actions hooked to `infinite_scroll_render` before the callback weren't firing. This would have introduced a bug into the implementation infinite scroll in P2, but it was never merged back to WPCOM. When I finally merged this back to WPCOM we introduced a bug into the P2 theme. 321-gh-p2tenberg.

#### Changes proposed in this Pull Request:

* Call the `infinite_scroll_render` action before executing callbacks so that any actions that have already been added are executed.

This shouldn't change any other existing behaviour, since we `break` once `$results['html']` is populated, but if other plugins rely on the current behaviour they might be affected.

#### Testing instructions:

* This might be easier to test on WPCOM
* Apply this patch to your sandbox
* Sandbox a P2
* Scroll down
* Check that the next posts load as expected
* Check that there aren't any additional empty posts at the top of the page

To test regressions:
- Sandbox scruffybentest20160808115427.wordpress.com
- Go to https://scruffybentest20160808115427.wordpress.com/blog/
- Scroll down and click on "Read more"
- Go to the "FIRST BLOG POST" and check that the map and slideshow blocks load correctly.

#### Proposed changelog entry for your changes:

* Fix infinite scroll bug

### CODE REVIEW

**Review by Copons — DISMISSED** [review]
I'm not very familiar with how infinite scroll works, but I've tested this on a P2 affected by the issue and it works for me now!

File: modules/infinite-scroll/infinity.php [hunk 1]
```diff
@@ -1329,40 +1329,45 @@ function query() {
...
-					 * @since 2.0.0
-					 */
-					do_action( 'infinite_scroll_render' );
-
-					$results['html'] = ob_get_clean();
-					remove_action( 'infinite_scroll_render', $callback );
-				}
-				if ( ! empty( $results['html'] ) ) {
-					break;
+			do_action( 'infinite_scroll_render' );
```
Comment (jeherve) [inline modules/infinite-scroll/infinity.php:1332]: The action is now triggered twice. Did you mean to add this one here?

**Review by jeherve — CHANGES_REQUESTED** [review]
This doesn't seem to be breaking things in my admittedly limited testing. We'd need to test it on a few Atomic sites with different themes / Infinite Scroll configurations to make sure that doesn't break anything.

I have small remark below.

File: modules/infinite-scroll/infinity.php (hunk 1 above)
Comment (scruffian) [reply modules/infinite-scroll/infinity.php:1332]: This is how it used to work, and the bug is created by removing this call. There might be a different way to fix the issue by changing how p2 itself works...

**Comment by scruffian** [comment]
I've tested this with every Twenty * theme on Atomic :)

**Review by lezama — DISMISSED** [review]
Thanks Ben for looking at this :)

**Review by jeherve — APPROVED** [review]
This seems to work well in my tests in Jetpack. 🚢 

Noting that I rebased and added a little comment for the next time we look at this code in a few years. :)

**Comment by jeherve** [comment]
r210108-wpcom

**Pull request merged**